features/api/services/applications/new.feature

```diff
@@ -0,0 +1,123 @@
+@javascript
+Feature: Product > Applications > New
+
+  Background:
+    Given a provider
+    And the provider has multiple applications enabled
+    And a product "My API"
+    And the following application plan:
+      | Product | Name  |
+      | My API  | Basic |
+    And a buyer "Jane"
+    And the following application:
+      | Buyer | Name   | Product |
+      | Jane  | My App | My API  |
+    And the provider logs in
+
+  Scenario: Navigation
+    Given they go to product "My API" applications page
+    When they follow "Create application"
+    Then the current page is product "My API" new application page
+
+  Scenario: Create an application
+    Given they go to product "My API" new application page
+    When the form is submitted with:
+      | Account          | Jane           |
+      | Application plan | Basic          |
+      | Name             | New App        |
+      | Description      | This is an app |
+    Then the current page is application "New App" admin page
+    And they should see the flash message "Application was successfully created"
+
+  Scenario: Create an application without being subscribed to a product
+    Given the buyer is not subscribed to product "My API"
+    And they go to product "My API" new application page
+    When the form is submitted with:
+      | Account          | Jane           |
+      | Application plan | Basic          |
+      | Name             | New App        |
+      | Description      | This is an app |
+    Then the current page is application "New App" admin page
+    And they should see the flash message "Application was successfully created"
+
+  Scenario: Create an application for a subscribed product
+    Given the buyer is subscribed to product "My API"
+    And they go to product "My API" new application page
+    When the form is submitted with:
+      | Account          | Jane           |
+      | Application plan | Basic          |
+      | Name             | New App        |
+      | Description      | This is an app |
+    Then the current page is application "New App" admin page
+    And they should see the flash message "Application was successfully created"
+
+  Scenario: Create an application when product has no application plans
+    Given a product "No plans API" with no plans
+    And they go to product "No plans API" new application page
+    Then they should see "No application plans exist for the selected product"
+    And select "Application plan" is disabled
+
+  Scenario: Create an application with a required extra field
+    Given the provider has the following fields defined for applications:
+      | name   | required | read_only | hidden |
+      | Banana | true     |           |        |
+    When they go to product "My API" new application page
+    And the form is submitted with:
```

Comment:
Here I'd rather verify that the form is not submittable without this extra field (because it's required)... But OK, nevermind :)

Reply:
Yes, totally agree! I'll fix it